festim/hydrogen_transport_problem.py

```diff
@@ -480,10 +481,39 @@ def create_dirichletbc_form(self, bc):
 
         return form
 
+    def define_sources(self):
```

Comment:
I wouldn't call this `define_sources` as the sources are already "defined". What this method does is create the fenics value object for the source.

Maybe also true for other methods like `define_boundary_conditions`. What do you think?